src/cmdstan/command.hpp

```diff
@@ -257,8 +257,8 @@ int command(int argc, const char *argv[]) {
     int num_paths = get_arg_val<int_argument>(*pathfinder_arg, "num_paths");
     bool save_iterations
         = (get_arg_val<bool_argument>(*pathfinder_arg, "save_single_paths")
-           || !get_arg_val<string_argument>(
-               parser, "output", "diagnostic_file").empty());
+           || !get_arg_val<string_argument>(parser, "output", "diagnostic_file")
+                   .empty());
```

Comment:
This change does fix the bug, but it also sets `save_iterations` for multipath-pathfinder, which will trigger all the single paths to _try_ to write diagnostics to streams we know are really nullptrs.

I think the right thing to do is to move the `save_iterations` logic inside the `if (num_paths == 1) {...} else {...}` statement

Reply:
doh!  of course - fixed.